src/api/component.js

```diff
@@ -189,6 +191,7 @@ Component.prototype = Object.create(HTMLElement.prototype, {
   createdCallback: {
     configurable: true,
     value() {
+      setElementAsDefined(this);
```

Comment:
I knew this was moved to `connectedCallback()`, but why do we need this here? Shouldn't it be causing tests in Canary to fail?